sparsity/sparse_frame.py

```diff
@@ -62,12 +62,19 @@ def __init__(self, data, index=None, columns=None, **kwargs):
 
         if index is None:
             self._index = _default_index(N)
+        elif len(index) != N and data.size:
+            raise ValueError('Shape of passed values is {},'
+                             'indices imply {}'
+                             .format(data.shape, (data.shape[0], len(index))))
```

Comment:
Firstly: should be `.format(data.shape, (len(index), data.shape[1]))` (similar error is below).

Secondly: if `columns` was also passed, then this error message will be slightly misleading (it would say that indices imply actual number of columns in data instead of number of columns passed). Same error is below.

I think this error message could be simpler and mention only one dimension. No need to solve this in smarter way.

Reply:
This is ananlogous to pandas error message.... I'll try to come up with something